### PR TITLE
Improve CMP HTTP test server

### PR DESCRIPTION
**Pull request by DDvO**
This is a further excerpt from #15053 to ease reviewing.

* CMP test server: Extend error reporting on cert rejected for revocation
* HTTP test server: Improve connection management and logging
* `cmp_server.c`: Improve transaction management and logging
* `OSSL_CMP_SRV_process_request()`: Log any error queue entries, which also increases stability

### CODE REVIEW

File: apps/lib/http_server.c [hunk 1]
```diff
@@ -266,19 +271,19 @@ int http_server_get_asn1_req(const ASN1_ITEM *it, ASN1_VALUE **preq,
...
+    int ret = 0;
 
     *preq = NULL;
     if (ppath != NULL)
         *ppath = NULL;
     *pcbio = NULL;
 
-    /* Connection loss before accept() is routine, ignore silently */
+    log_message(prog, LOG_DEBUG, "Awaiting next request...");
+/* Connection loss before accept() is routine, ignore silently */
```
Comment (t8m) [inline apps/lib/http_server.c:282]: Why the indentation change?

**Review by t8m — APPROVED** [review]
LGTM. One trivial nit fixable when merging.

File: apps/lib/http_server.c (hunk 1 above)
Comment (DDvO) [reply apps/lib/http_server.c:282]: This indentation change was by mistake.
It will be changed (with correct indentation) by #15053.

**Comment by DDvO** [comment]
This PR is meanwhile superseded by #15053, which has been approved as well.
So closing this excerpt, which was provided to simplify reviewing, has become obsolete.

**Pull request closed without merging**